sv_pipeline/genome/load_data.py

```diff
@@ -0,0 +1,185 @@
+    'geneIds': lambda rows: rows.gene_affected.filter(lambda x: x[1] != 'NEAREST_TSS').flatmap(lambda x: x[0]),
+    'samples_num_alt_0': lambda rows: get_sample_num_alt_x(rows, 0),
+    'samples_num_alt_1': lambda rows: get_sample_num_alt_x(rows, 1),
+    'samples_num_alt_2': lambda rows: get_sample_num_alt_x(rows, 2),
+}
+
+FIELDS = list(BASIC_FIELDS.keys()) + list(COMPUTED_FIELDS.keys()) + ['genotypes']
+
+STR_ARRAY_FIELDS = {'filters', 'geneIds', 'samples_num_alt_0', 'samples_num_alt_1', 'samples_num_alt_2'}
+
+
+def get_sample_num_alt_x(rows, n):
+    return rows.genotypes.filter(lambda x: x.num_alt == n).map(lambda x: x.sample_id)
+
+
+def sub_setting_mt(project_guid, mt, sample_type, skip_sample_subset, ignore_missing_samples):
+    found_samples = {col.s for col in mt.cols().collect()}
+    if skip_sample_subset:
+        sample_subset = found_samples
+    else:
+        sample_subset = get_sample_subset(project_guid, sample_type)
+        if len(found_samples) != len(sample_subset):
+            missed_samples = sample_subset - found_samples
+            missing_sample_error = 'Missing the following {} samples:\n{}'.format(
+                len(missed_samples), ', '.join(sorted(missed_samples))
+            )
+            if ignore_missing_samples:
+                logger.info(missing_sample_error)
+            else:
+                skipped_samples = found_samples - sample_subset
+                logger.info('Samples in callset but skipped:\n{}'.format(', '.join(sorted(skipped_samples))))
+                raise Exception(missing_sample_error)
+        sample_remap = get_sample_remap(project_guid, sample_type)
+        message = 'Subsetting to {} samples'.format(len(sample_subset))
+        if sample_remap:
+            message += ' (remapping {} samples)'.format(len(sample_remap))
+            sample_subset = sample_subset - set(sample_remap.keys()) + set(sample_remap.values())
+            mt = mt.annotate_cols(**sample_remap)
+        logger.info(message)
+
+    mt = mt.filter_cols(hl.literal(sample_subset).contains(mt['s']))
+
+    genotypes = hl.agg.collect(hl.struct(sample_id=mt.s, gq=mt.GQ, num_alt=mt.GT.n_alt_alleles(), cn=mt.RD_CN))
+    mt = mt.annotate_rows(genotypes=genotypes)
+    return mt.filter_rows(mt.genotypes.any(lambda x: x.num_alt > 0)).rows()
+
+
+def annotate_fields(rows):
+    rows = rows.annotate(**{k: v(rows) for k, v in BASIC_FIELDS.items()})
+
+    rows = rows.annotate(gene_affected=hl.filter(lambda x: hl.is_defined(x[0]),
+                              [(rows.info[col], col.split('__')[-1]) for col in
+                               [gene_col for gene_col in rows.info if gene_col.startswith('PROTEIN_CODING__')
+                                and rows.info[gene_col].dtype == hl.dtype('array<str>')]]))
+
+    rows = rows.annotate(**{k: v(rows) for k, v in COMPUTED_FIELDS.items()})
+
+    # replace empty string array fields with nulls
```

Comment:
as I mentioned in the previous PR review, you only need to do this for the `filters` field. It might honestly be neater to do it as a computed field where you add the field in the basic fields and then override it with an `if_else` in the computed fields